src/Features/Core/Portable/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs

```diff
@@ -0,0 +1,111 @@
+                context.Diagnostics);
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private async Task FixWithEditorAsync(
+            Document document, SyntaxEditor editor, ImmutableArray<Diagnostic> diagnostics,
+            CancellationToken cancellationToken)
+        {
+            var declarators = new List<TSymbolSyntax>();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+                declarators.Add(root.FindNode(diagnosticSpan).FirstAncestorOrSelf<TSymbolSyntax>());
+            }
+
+            MakeFieldReadonly(document, editor, declarators);
+        }
+
+        private async void MakeFieldReadonly(Document document, SyntaxEditor editor, List<TSymbolSyntax> declarators)
+        {
+            var declaratorsByField = declarators.GroupBy(g => g.FirstAncestorOrSelf<TFieldDeclarationSyntax>());
+
+            foreach (var fieldDeclarators in declaratorsByField)
+            {
+                var declarationDeclarators = GetVariableDeclarators(fieldDeclarators.Key);
+
+                if (declarationDeclarators.Count == fieldDeclarators.Count())
+                {
+                    editor.SetModifiers(fieldDeclarators.Key, editor.Generator.GetModifiers(fieldDeclarators.Key) | DeclarationModifiers.ReadOnly);
+                }
+                else
+                {
+                    var model = await document.GetSemanticModelAsync().ConfigureAwait(false);
+                    var generator = editor.Generator;
+                    
+                    foreach (var declarator in declarationDeclarators.Reverse())
+                    {
+                        var symbol = (IFieldSymbol)model.GetDeclaredSymbol(declarator);
+                        var modifiers = generator.GetModifiers(fieldDeclarators.Key);
+
+                        var newDeclaration = generator.FieldDeclaration(symbol.Name,
+                                                                        generator.TypeExpression(symbol.Type),
+                                                                        Accessibility.Private,
+                                                                        fieldDeclarators.Contains(declarator)
+                                                                            ? modifiers | DeclarationModifiers.ReadOnly
+                                                                            : modifiers,
+                                                                        GetInitializerNode(declarator))
+                                                      .WithAdditionalAnnotations(Formatter.Annotation);
+
+                        editor.InsertAfter(fieldDeclarators.Key, newDeclaration);
+                    }
+
+                    editor.RemoveNode(fieldDeclarators.Key);
```

Comment:
do we have fix-all tests that remove multiple declarators from a declaration (but not all).  I just want to make sure that works.  i.e. have 3 variable decls in the field, then have 2 of them pulled out while one remains.

(also, all the crazy VB cases).

Reply:
➡️ Added in commit 38db528 (One such VB case already exists, others are treated separately)